packages/patternfly-3/patternfly-react/src/components/LoginPage/components/LoginCardComponents/LoginCardInput.js

```diff
@@ -23,28 +22,30 @@ const LoginCardInput = ({
   autoComplete,
   attributes
 }) => {
-  const helpBlock =
-    (showError && <HelpBlock>{error}</HelpBlock>) ||
-    (showWarning && <LoginCardInputWarning>{warning}</LoginCardInputWarning>);
-  const validationState = showError ? 'error' : null;
+  // eslint-disable-next-line no-nested-ternary
+  const validationState = showError ? 'error' : showWarning ? 'warning' : null;
+  const tooltip = (
+    <Tooltip id={`${id}-tooltip`} style={showWarning || showError ? '' : { display: 'None' }}>
+      {showWarning ? warning : error}
+    </Tooltip>
+  );
   return (
-    <FormGroup className={classNames('login_card_input', className)} controlId={id} validationState={validationState}>
-      <FormControl
-        {...attributes}
-        type={type}
-        placeholder={placeholder}
-        bsSize={size}
-        onChange={onChange}
-        onFocus={onFocus}
-        onBlur={onBlur}
-        onKeyPress={onKeyPress}
-        autoComplete={autoComplete}
-      />
-
-      <Fade in={showError || showWarning}>
-        <div>{helpBlock}</div>
-      </Fade>
-    </FormGroup>
+    <OverlayTrigger overlay={tooltip} placement="right">
```

Comment:
I think that if we will use in this case `Overlay` instead of `OverlayTrigger` 
we will have more control on the tooltip visibillity and position.
It's a bit more work but in my opinion it will be much easier to maintain.